model/model.go

```diff
@@ -0,0 +1,131 @@
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Operator common.Address `gorm:"column:operator;uniqueIndex:idx_operator_date"`
+	// RewardAfterCommission query StakeCredit rewardRecord at the end of the day
+	RewardAfterCommission *Big `gorm:"column:reward_after_commission;type:VARBINARY(32)"`
+	// TotalPooledBNB query StakeCredit totalPooledBNBRecord at the end of the day
+	TotalPooledBNB *Big `gorm:"column:total_pooled_bnb;type:VARBINARY(32)"`
+	// TotalCreditToken query StakeCredit _totalSupply at the previous block of the breathing block
+	TotalCreditToken *Big `gorm:"column:total_credit_token;type:VARBINARY(32)"`
+
+	Date int64 `gorm:"column:date;uniqueIndex:idx_operator_date"`
+}
+
+// Delegator is a model for the delegator table.
+// only record their delegator from config.
+// need listen stakeHub Delegated event?
+type Delegator struct {
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Delegator common.Address `gorm:"column:delegator"`
+	Operator  common.Address `gorm:"column:operator"`
+	// Amount query stakeCredit getPooledBNB interface at the previous block of the breathing block
+	Amount *Big `gorm:"column:amount;type:VARBINARY(32)"`
+
+	Date int64 `gorm:"column:date;index:idx_date"`
+}
+
+type DelegateAction string
+
+const (
+	Delegate    DelegateAction = "delegate"
+	UnDelegate  DelegateAction = "undelegate"
+	ReDelegator DelegateAction = "redelegate"
+)
+
+// DelegateTx is a model for the delegate_tx table.
+// getLogs{toAddress:stakeHub, topic:Delegated/Undelegated/Redelegated}.
+type DelegateTx struct {
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Delegator   common.Address `gorm:"column:delegator"`
+	SrcOperator common.Address `gorm:"column:src_operator"`
+	DstOperator common.Address `gorm:"column:dst_operator"`
+	Action      DelegateAction `gorm:"column:action"`
+	Amount      *Big           `gorm:"column:amount;type:VARBINARY(32)"`
+	TxHash      common.Hash    `gorm:"column:tx_hash;uniqueIndex:idx_tx_hash"`
+}
+
+// BreathBlockRewardEvent
+// getLogs{toAddress:[credits], topics:RewardReceived}
+type BreathBlockRewardEvent struct {
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Credit                common.Address `gorm:"column:operator;uniqueIndex:idx_credit_date"`
+	RewardAfterCommission *Big           `gorm:"column:reward_after_commission;type:VARBINARY(32)"`
+	Commission            *Big           `gorm:"column:commission;type:VARBINARY(32)"`
+
+	Date int64 `gorm:"column:date;uniqueIndex:idx_credit_date"`
+}
+
+// SlashEvent
+// getLogs{toAddress:[stakeHub], topics:ValidatorSlashed}
+type SlashEvent struct {
+	ID uint64 `gorm:"column:id;primaryKey"`
+
+	Operator common.Address `gorm:"column:operator;index:idx_operator"`
+	Amount   *Big           `gorm:"column:amount;type:VARBINARY(32)"`
+	TxHash   common.Hash    `gorm:"column:tx_hash;uniqueIndex:idx_tx_hash"`
+
+	Date int64 `gorm:"column:date;index:idx_date"`
+}
+
+// Header is a model for the header table.
+type Header struct {
+	ID uint64 `gorm:"column:id;primaryKey" rlp:"-"`
```

Comment:
do we really need this table.. I thought just a table to record to where our service have synced to.

Reply:
done